citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -211,7 +211,18 @@ public ICitation updateCitationFromZotero(IUser user, String groupId, String ite
             throws GroupDoesNotExistException, CannotFindCitationException, ZoteroHttpStatusException {
         Optional<ICitationGroup> groupOptional = groupRepository.findFirstByGroupId(new Long(groupId));
         if (!groupOptional.isPresent()) {
-            throw new GroupDoesNotExistException("Group with id " + groupId + " does not exist.");
+            ICitationGroup group = groupManager.getGroup(user, groupId);
+            if (group == null) {
+                throw new GroupDoesNotExistException("Group with id " + groupId + " does not exist.");
+            } else {
+                asyncCitationProcessor.sync(user, groupId, group.getContentVersion(), null);
+                try {
+                    ICitation citation = zoteroManager.getGroupItem(user, groupId, itemKey);
+                    return citation;
+                } catch (HttpClientErrorException ex) {
+                    throw new CannotFindCitationException(ex);
+                }
```

Comment:
I think you need the below code when a citation is retrieved. Is there a reason you return here instead of just starting the sync and then keep going below?